addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/EncoderUtils.java

```diff
@@ -0,0 +1,302 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.grpc.internal;
+
+import com.google.protobuf.CodedOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EncoderUtils {
+
+    public static final int VARINT_WIRE_TYPE = 0;
+    public static final int BIT64_WIRE_TYPE = 1;
+    public static final int LENGTH_DELIMITED_WIRE_TYPE = 2;
+    public static final int BIT32_WIRE_TYPE = 5;
+
+    public static String removeFirstAndLastCurlyBraces(String text)
+            throws InvalidProtobufFormatException {
+        // Check if the string starts with '{' and ends with '}' for nested message
+        if (text != null
+                && text.length() >= 2
+                && text.charAt(0) == '{'
+                && text.charAt(text.length() - 1) == '}') {
+            // Remove the first and last characters using substring
+            int first = text.indexOf('\n');
+            int last = text.lastIndexOf('\n');
+            if (first == -1 || last == -1)
+                throw new InvalidProtobufFormatException(
+                        "Invalid Format:Nested message must contain a newline immediately after every curly brace");
+            return text.substring(first + 1, last);
+        } else {
+            throw new InvalidProtobufFormatException(
+                    "Invalid Format: Nested message should start and end with curly braces");
+        }
+    }
+
+    public static List<String> parseIntoList(String inputString)
+            throws InvalidProtobufFormatException {
+        if (inputString == null || inputString.isEmpty()) {
+            return new ArrayList<>();
+        }
+        List<String> output = new ArrayList<>();
+        String[] temp = inputString.split("\n");
+
+        StringBuilder field = new StringBuilder();
+        int countOpenCurlyBraces = 0;
+
+        for (String s : temp) {
+            if (s.charAt(s.length() - 1) != '{'
+                    && s.charAt(0) != '}'
+                    && countOpenCurlyBraces == 0) {
+                output.add(s);
+            } else if (s.charAt(s.length() - 1) == '{') {
+                countOpenCurlyBraces++;
+                field.append(s).append('\n');
+            } else if (s.charAt(0) == '}') {
+                countOpenCurlyBraces--;
+                if (countOpenCurlyBraces == 0) {
+                    field.append(s);
+                    output.add(field.toString());
+                    field.setLength(0);
+                } else {
+                    field.append(s).append('\n');
+                }
+            } else field.append(s).append('\n');
```

Comment:
This is syntactically correct but should also use braces to maintain consistency.

Reply:
No, I mean on this else statement, it should be wrapped like the previous one.

Reply:
Here still